client/packages/programs/src/VaccineCourseEditModal/VaccineCourseEditModal.tsx

```diff
@@ -217,3 +216,125 @@ export const VaccineCourseEditModal: FC<VaccineCourseEditModalProps> = ({
+
+  const addDose = () => {
+    updatePatch({
+      vaccineCourseSchedules: [
+        ...doses,
+        {
+          __typename: 'VaccineCourseScheduleNode',
+          id: FnUtils.generateUUID(),
+          // temp - will be overwritten by the backend to assign unique dose number (even if previous doses were deleted)
+          doseNumber: doses.length + 1,
+          label: `${courseName} ${doses.length + 1}`,
+          minAgeMonths: 0,
+          minIntervalDays: 0,
+        },
+      ],
+    });
+  };
+
+  const deleteDose = (id: string) => {
+    updatePatch({
+      vaccineCourseSchedules: doses.filter(dose => dose.id !== id),
+    });
+  };
+
+  const updateDose = (
+    id: string,
+    newData: Partial<VaccineCourseScheduleNode>
+  ) => {
+    updatePatch({
+      vaccineCourseSchedules: doses.map(dose =>
+        dose.id === id ? { ...dose, ...newData } : dose
+      ),
+    });
+  };
+
+  return (
+    <>
+      <Box display={'flex'} justifyContent={'flex-end'}>
+        <ButtonWithIcon
+          Icon={<PlusCircleIcon />}
+          label={t('label.dose')}
+          onClick={addDose}
+        />
+      </Box>
+      <Table sx={{ marginTop: '5px', '& td': { padding: '3px' } }}>
+        <tr style={{ borderBottom: '1px solid lightgray' }}>
+          <HeaderCell label="Dose #" width="80px" />
+          <HeaderCell label={t('label.label')} width="250px" />
+          <HeaderCell label="Age (months)" />
+          <HeaderCell label="Min interval (days)" />
+        </tr>
+        <tbody>
+          {doses.map((dose, index) => (
+            <tr>
+              <td style={{ textAlign: 'center' }}>{index + 1}</td>
+              <td>
+                <BasicTextInput
+                  value={dose.label}
+                  fullWidth
+                  onChange={e => updateDose(dose.id, { label: e.target.value })}
+                />
+              </td>
+              <NumericDoseTableCell
+                value={dose.minAgeMonths}
+                onChange={num => updateDose(dose.id, { minAgeMonths: num })}
+              />
+              <NumericDoseTableCell
+                value={dose.minIntervalDays}
+                onChange={num => updateDose(dose.id, { minIntervalDays: num })}
+              />
+              <td style={{ display: 'flex', justifyContent: 'center' }}>
+                <IconButton
+                  icon={<DeleteIcon />}
+                  label={t('label.delete')}
+                  onClick={() => deleteDose(dose.id)}
+                />
+              </td>
+            </tr>
+          ))}
+        </tbody>
+      </Table>
+    </>
+  );
+};
+
+const HeaderCell = ({ label, width }: { label: string; width?: string }) => {
+  return <th style={{ fontSize: '14px', width, padding: '3px' }}>{label}</th>;
+};
+
+const NumericDoseTableCell = ({
+  value,
+  onChange,
+}: {
+  value: number;
+  onChange: (num: number) => void;
+}) => {
+  const [buffer, setBuffer] = useBufferState<number | undefined>(value);
```

Comment:
I thought we were trying to move away from using this hook these days? Is it necessary?

Also, this component seems pretty generic -- can we just use the existing `NumberInputCell`? (There doesn't seem be anything specific to this particular use case here)

Reply:
Ah yeah you're right, had some other logic in here but didn't end up needing it... Didn't use the `NumberInputCell` because wasn't using our usual DataTable component, but realised I probably should have! Updated [6d02161](https://github.com/msupply-foundation/open-msupply/pull/4717/commits/6d021613ce812a31627bd0e33bde52e0640cfa4b) 😁

Reply:
Hmm, looks-wise I think it was better how you had it. I don't think the DataTable UI suits a small space like this. Not sure what to suggest. Quick chat at tomorrow stand-up?